runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsPlugin.java

```diff
@@ -252,22 +253,30 @@ public void onMenuOptionClicked(MenuOptionClicked event)
 			String itemName = itemComposition.getName();
 			String initialValue = tagManager.getTagString(itemId);
 
-			chatboxInputManager.openInputWindow(itemName + " tags:", initialValue, (newTags) ->
+			boolean isSearchOpen = client.getVar(VarClientInt.INPUT_TYPE) == InputType.SEARCH.getType();
+			String searchText = client.getVar(VarClientStr.INPUT_TEXT);
+
+			chatboxPanelManager.openTextInput(itemName + " tags:")
+				.value(initialValue)
+				.onDone((newTags) ->
+					clientThread.invoke(() ->
+					{
+						tagManager.setTagString(itemId, newTags);
+
+						// Check both previous and current tags in case the tag got removed in new tags or in case
+						// the tag got added in new tags
+						final List<String> initialTags = SPLITTER.splitToList(initialValue.toLowerCase());
+						final List<String> tags = SPLITTER.splitToList(newTags.toLowerCase());
+						tabInterface.updateTabIfActive(initialTags);
+						tabInterface.updateTabIfActive(tags);
+					}))
+				.build();
+
+			if (isSearchOpen)
```

Comment:
Why is this necessary now?

Reply:
You have bad habit of putting everything to 1 PR without explanation. You should at least split that to separate commit and reference that issue